functional-tests/appwrapper_reconciler_test.go

```diff
@@ -0,0 +1,222 @@
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "project-codeflare", "multi-cluster-app-dispatcher@v1.38.1", "config", "crd", "bases"),
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "openshift", "api@v0.0.0-20220411210816-c3bb724c282a", "machine", "v1beta1"),
+		},
+	}
+	cfg, err = testEnv.Start()
+	test.Expect(err).NotTo(HaveOccurred())
+
+	return testEnv
+}
+
+func establishClient(t *testing.T) {
+	test := With(t)
+	err = mcadv1beta1.AddToScheme(scheme.Scheme)
+	err = machinev1beta1.AddToScheme(scheme.Scheme)
+	test.Expect(err).NotTo(HaveOccurred())
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	test.Expect(err).NotTo(HaveOccurred())
+	test.Expect(k8sClient).NotTo(BeNil())
+
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
+	test.Expect(err).ToNot(HaveOccurred())
+
+	instaScaleController := &controllers.AppWrapperReconciler{
+		Client: k8sManager.GetClient(),
+		Scheme: k8sManager.GetScheme(),
+		Config: config.InstaScaleConfiguration{
+			MachineSetsStrategy: "reuse",
+			MaxScaleoutAllowed:  5,
+		},
+	}
+	err = instaScaleController.SetupWithManager(context.Background(), k8sManager)
+	test.Expect(err).ToNot(HaveOccurred())
+
+	go func() {
+		err = k8sManager.Start(ctrl.SetupSignalHandler())
+		test.Expect(err).ToNot(HaveOccurred())
+	}()
+
+}
+
+func teardownTestEnv(testEnv *envtest.Environment) {
+	logger := ctrl.LoggerFrom(ctx)
+	if err := testEnv.Stop(); err != nil {
+		logger.Error(err, "Error stopping test Environment\n")
+	}
+}
+
+func instascaleAppwrapper(namespace string) *mcadv1beta1.AppWrapper {
+	aw := &mcadv1beta1.AppWrapper{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-instascale",
+			Namespace: namespace,
+			Labels: map[string]string{
+				"orderedinstance": "test.instance1",
+			},
+			Finalizers: []string{"instascale.codeflare.dev/finalizer"},
+		},
+		Spec: mcadv1beta1.AppWrapperSpec{
+			AggrResources: mcadv1beta1.AppWrapperResourceList{
+				GenericItems: []mcadv1beta1.AppWrapperGenericResource{
+					{
+						DesiredAvailable: 1,
+						CustomPodResources: []mcadv1beta1.CustomPodResourceTemplate{
+							{
+								Replicas: 1,
+								Requests: apiv1.ResourceList{
+									apiv1.ResourceCPU:    resource.MustParse("250m"),
+									apiv1.ResourceMemory: resource.MustParse("512Mi"),
+								},
+								Limits: apiv1.ResourceList{
+									apiv1.ResourceCPU:    resource.MustParse("1"),
+									apiv1.ResourceMemory: resource.MustParse("1G"),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return aw
+}
+
+func TestReconciler(t *testing.T) {
+	testEnv = startEnvTest(t)      // initiates setting up the EnvTest environment
+	defer teardownTestEnv(testEnv) //defer teardown of test environement until function exists
+
+	establishClient(t) // setup client required for test
+
+	test := With(t)
+
+	//create new test namespace
+	namespace := test.NewTestNamespace()
```

Comment:
This Appears to be having an issue with the communicating with our client, and is throwing the following error when run with the envtest environment:
```
Error creating client: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
```

I might suggest creating a new namespace directly with if we can't find the root of it:

```go
	namespace := &apiv1.Namespace{
		ObjectMeta: metav1.ObjectMeta{
			Name: "test-namespace",
		},
	}
```